third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1788,34 +1333,24 @@ impl<'a> Resolver<'a> {
             BinaryType::Module(module) => &module.field_instantiations[idx.0 as usize],
             BinaryType::Script(_) => unreachable!("Scripts cannot have type instructions"),
         };
-        let struct_type = self
-            .loader
-            .get_struct_type(field_instantiation.owner)
-            .ok_or_else(|| {
-                PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
-                    .with_message("Struct Definition not resolved".to_string())
-            })?;
 
         let instantiation_types = field_instantiation
             .instantiation
             .iter()
             .map(|inst_ty| inst_ty.subst(ty_args))
             .collect::<PartialVMResult<Vec<_>>>()?;
-        struct_type.fields[field_instantiation.offset].subst(&instantiation_types)
+        field_instantiation.definition_struct_type.fields[field_instantiation.offset]
+            .subst(&instantiation_types)
     }
 
     pub(crate) fn get_struct_fields(
```

Comment:
and this one is called get fields and returns StructType...

Reply:
+1